pygsti/algorithms/directx.py

```diff
@@ -394,7 +395,9 @@ def direct_mc2gst_models(circuits, dataset, prep_fiducials, meas_fiducials,
         for i, sigma in enumerate(circuits):
             printer.show_progress(i, len(circuits), prefix="--- Computing model for string-", suffix='---')
             directLSGSTmodels[sigma] = direct_mc2gst_model(
-                sigma, "GsigmaLbl", dataset, prep_fiducials, meas_fiducials, target_model,
+                sigma, 
+                Label('GsigmaLbl') if sigma.line_labels == ('*',) else Label('GsigmaLbl', sigma.line_labels), 
+                dataset, prep_fiducials, meas_fiducials, target_model,
```

Comment:
Interesting that you've made a change to this file, since Erik just marked it as minimal-exposure. @coreyostrove any chance you'd disagree with Erik's take on this file's user exposure? Here's the conventions we agreed to, for reference:
* High exposure: API changes are likely to break workflows for some users, including "casual" users.
* Low exposure: API changes are unlikely to affect casual users of pyGSTi but might affect some of pyGSTI's advanced users.
* Minimal exposure: while it's conceivable that API changes could break user code, the possibility of this is sufficiently remote that we can change the API at will.

Reply:
Good question. I agree with Erik's marking this as minimal-exposure. This change arose as a bugfix for some failures encountered in the corresponding test package (test_directx.py) as part of the refactor moving from legacy modelpacks to the modern implementation in testing. This particular line of code turned out to be fragile when it came to circuit line labeling, and so broke when using circuits from the new modelpacks that have different default circuit line labeling behavior than they do in the legacy modelpacks. 

In reality I suspect this is likely due to a disjoint bug in the circuit line label handling code itself in the circuits module that I've run into in other contexts and mentioned to @sserita before, but haven't ever properly documented. (This is a bug related to properly reconciling line labels between circuits with a  placeholder '*' and those with fully specified labels). In the moment it was easier to patch in a workaround in directx than to context switch and try to track the circuit bug. That said, I am in the correct headspace for this sort of thing now, so if nothing else I should be able to get a minimal reproduction of the aforementioned bug documented pretty soon.

While I was working in the directx code I did wonder to myself if we even need any of this anymore? I've personally never interacted with this code in practical usage, and don't know of anyone who does use this code. Given that, maybe this would be a decent candidate for a module we can retire to legacy as part of out spring cleaning process. @enielse and @sserita, thoughts on this last question?

Reply:
Update: I just read through @enielse's latest commit on the code marking branch and saw that he has marked directx as:

> ## user-exposure: minimal, forsaken (EGN - we don't use this anymore and never really did)

So that is two votes in favor of decommissioning this part of the codebase. @sserita, how do you feel about formally marking this as deprecated in 0.9.12.0 and then removing it (remove = move to the legacy folder) either in 0.9.13.0 or some point thereafter?

Reply:
I agree with the minimal tag and we can formally mark this as deprecated. I will do so in the bugfix-for-0.9.12 branch before I merge that in.